helper/helper/src/main/java/com/linkedin/avroutil1/compatibility/CompatibleSpecificRecordBuilderBase.java

```diff
@@ -48,7 +48,7 @@ protected final boolean[] fieldSetFlags() {
    *                              accept null values.
    */
   protected void validate(Schema.Field field, Object value) {
-    if (!isValidValue(field, value) && field.defaultValue() == null) {
+    if (!isValidValue(field, value) && AvroCompatibilityHelper.getSpecificDefaultValue(field) == null) {
```

Comment:
nit: do we have a simple test cover this?

Reply:
Added.